lightningd/offer.c

```diff
@@ -370,23 +371,49 @@ static bool payer_key(struct lightningd *ld,
 					     tweakhash.u.u8) == 1;
 }
 
+static void json_populate_invreq(struct json_stream *response,
+				 const struct sha256 *invreq_id,
+				 const char *b12,
+				 const struct json_escape *label,
+				 enum offer_status status)
+{
+	json_add_sha256(response, "invreq_id", invreq_id);
+	json_add_bool(response, "active", offer_status_active(status));
+	json_add_bool(response, "single_use", offer_status_single(status));
+	json_add_string(response, "bolt12", b12);
+	json_add_bool(response, "used", offer_status_used(status));
+	if (label)
+		json_add_escaped_string(response, "label", label);
+}
+
 static struct command_result *json_createinvoicerequest(struct command *cmd,
 							const char *buffer,
 							const jsmntok_t *obj,
 							const jsmntok_t *params)
 {
 	struct tlv_invoice_request *invreq;
-	const char *label;
+	struct json_escape *label;
 	struct json_stream *response;
 	u64 *prev_basetime = NULL;
 	struct sha256 merkle;
+	bool *save, *single_use;
+	enum offer_status status;
+	struct sha256 invreq_id;
+	const char *b12str;
 
 	if (!param(cmd, buffer, params,
 		   p_req("bolt12", param_b12_invreq, &invreq),
-		   p_opt("recurrence_label", param_escaped_string, &label),
+		   p_req("savetodb", param_bool, &save),
```

Comment:
Is it that common to create an invoice request, but then not save it? Wondering if a `p_opt_def` defaulting to `true` wouldn't be more ergonomic.

Reply:
It's actually most common *not* to save it; you only save it if you're going to publish it (e.g. ATM or refund flow).  Otherwise, you're just creating it (really, populating the signature field) to request an invoice for a specific offer.

Hence the weird backward default...